flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/IntermediateResultInfo.java

```diff
@@ -35,6 +35,14 @@ public interface IntermediateResultInfo {
      */
     boolean isBroadcast();
 
+    /**
+     * Indicates whether every downstream consumer needs to consume all produced sub-partitions.
+     *
+     * @return true if every downstream consumer needs to consume all produced sub-partitions, false
+     *     otherwise.
+     */
+    boolean isEveryConsumerConsumeAllSubPartitions();
```

Comment:
Is this used to distinguish whether it is a predefined broadcast or a later converted broadcast result?